src/WidgetRegistry.ts

```diff
@@ -61,15 +68,16 @@ export class WidgetRegistry extends Evented implements WidgetRegistry {
 	/**
 	 * internal map of labels and WidgetRegistryItem
 	 */
-	private registry: Map<string, WidgetRegistryItem> = new Map<string, WidgetRegistryItem>();
+	private registry: Map<RegistryLabel, WidgetRegistryItem> = new Map<RegistryLabel, WidgetRegistryItem>();
 
-	has(widgetLabel: string): boolean {
+	has(widgetLabel: RegistryLabel): boolean {
 		return this.registry.has(widgetLabel);
 	}
 
-	define(widgetLabel: string, item: WidgetRegistryItem): void {
+	define(widgetLabel: RegistryLabel, item: WidgetRegistryItem): void {
 		if (this.registry.has(widgetLabel)) {
-			throw new Error(`widget has already been registered for '${widgetLabel}'`);
+			const regsitryLabelDesc = typeof widgetLabel === 'string' ? widgetLabel : widgetLabel.toString();
```

Comment:
not all the branches of this are being tested, but `'foo'.toString()` is valid syntax, so it maybe just easier to do that or use explicit coercion of `String(widgetLabel)`;